### PR TITLE
Read email from email_address but not from user table

### PR DESCRIPTION
**Pull request by lunny**
Since the `email` column in `User` will be dropped some day. We should only get email information from `EmailAddress`.

### CODE REVIEW

**Comment by wxiaoguang** [comment]
I have a question about EmailAddress. What's the purpose of `IsPrimary`?

`GetUserByEmail` only checks `IsActivated: true`.

Maybe we need more comments.

**Comment by lunny** [comment]
> I have a question about EmailAddress. What's the purpose of `IsPrimary`?
> 
> `GetUserByEmail` only checks `IsActivated: true`.
> 
> Maybe we need more comments.

OK. I will add more comments on Code.
`IsPrimary` means the email address will be used as the first one.
`IsActivated` means the email address has been confirmed.

File: services/auth/signin.go [hunk 1]
```diff
@@ -23,31 +21,26 @@ import (
...
-			return nil, nil, models.ErrUserNotExist{Name: username}
-		}
-
 		user = &models.User{LowerName: strings.ToLower(trimmedUsername)}
-	}
-
-	hasUser, err := models.GetUser(user)
-	if err != nil {
-		return nil, nil, err
+		hasUser, err = models.GetUser(user)
```
Comment (wxiaoguang) [inline services/auth/signin.go]: Do we need a `GetUserByName` like `GetUserByEmail`?

Then the username can be prepared (trimed, lower-cased) correctly inside this function.

Comment (lunny) [reply services/auth/signin.go]: I think we already have `GetUserByName`.

Comment (wxiaoguang) [reply services/auth/signin.go]: Why don't we use it?

Comment (lunny) [reply services/auth/signin.go]: done.

File: routers/web/user/auth.go [hunk 2]
```diff
@@ -1281,13 +1281,11 @@ func createUserInContext(ctx *context.Context, tpl base.TplName, form interface{
 	if err := user_model.CreateUser(u); err != nil {
 		if allowLink && (user_model.IsErrUserAlreadyExist(err) || user_model.IsErrEmailAlreadyUsed(err)) {
 			if setting.OAuth2Client.AccountLinking == setting.OAuth2AccountLinkingAuto {
-				var user *user_model.User
-				user = &user_model.User{Name: u.Name}
+				var user = &user_model.User{Name: u.Name}
```
Comment (wxiaoguang) [inline routers/web/user/auth.go]: Do we need `GetUserByName` here?

**Review by wxiaoguang — APPROVED** [review]

**Pull request closed without merging**